packages/e2e/test/artillery/StakePoolSearch.ts

```diff
@@ -0,0 +1,155 @@
+import * as envalid from 'envalid';
+import { ArtilleryContext, FunctionHook, WhileTrueHook } from './artillery';
+import { Cardano, Paginated, QueryStakePoolsArgs } from '@cardano-sdk/core';
+import { logger } from '@cardano-sdk/util-dev';
+import { stakePoolHttpProvider } from '@cardano-sdk/cardano-services-client';
+
+/**
+ * The context variables shared between all the hooks.
+ * Hooks must record here the state of the session of current artillery virtual user
+ */
+interface StakePoolSearchVars extends Paginated<Cardano.StakePool> {
+  /**
+   * The arguments used for the query; to repeat the same query, but next page, on next iteration
+   */
+  args: QueryStakePoolsArgs;
+
+  /**
+   * `performQuery()` stores here if it gets an error from the query to let `moreResults()` knows when exit on error
+   */
+  gotError?: boolean;
+
+  /**
+   * Tricky array which contains `1` element when the healthCheck is ok, `0` elements otherwise.
+   *
+   * The most _artillery native_ found way to implement logic `if(healthCheck is ok) perform the test;`
+   */
+  healthCheckArray: number[];
```

Comment:
```suggestion
   * The most _artillery native_ found way to implement logic `if(healthCheck is ok) perform the test;`
   */
  emptyWhenOver: number[];
```

This is quite odd. Are there alternatives?

Reply:
Hi @mkazlauskas ,

this was mainly an attempt to make a most _artillery native implementation_ as possible.

A possible alternative could be:
- to store the result of the health check in a `boolean` (rather than in a `number[]`)
- to check the result of the health check at the beginning of `performQuery` and completely skipping the query in case of provider unhealthy

This would make the artillery yaml pseudocode more readable

from:
```
scenarios:
  - name: 'Browse'
    flow:
      - function: 'healthCheck'
      - loop:
          - loop:
              - function: 'performQuery'
            whileTrue: 'moreResults'
        over: 'healthCheckArray'
```

to:
```
scenarios:
  - name: 'Browse'
    flow:
      - function: 'healthCheck'
      - loop:
          - function: 'performQuery'
        whileTrue: 'moreResults'
```

(more or less...)

Reply:
If we could still use `over:` with
> to store the result of the health check in a boolean (rather than in a number[])

then I think that would be easier to understand and also to name the property :sweat_smile:

Reply:
1185fc810 should solve this